test/test-errortracker.js

```diff
@@ -249,18 +246,118 @@ describe('Test how server responds to requests/behave', function() {
-    query.s = 'exception';
+    query.s = '  at new vi (https://cdn.ampproject.org/rtv/031496877433269/v0.js:297:149)';
     query.debug = 0;
     query.r = 'referer';
     query.m = 'message';
     return chai.request(app).get('/r').query(query).then(function(res) {
       expect(res).to.have.status(statusCodes.NO_CONTENT);
     });
   });
+
+  it('Should not log exceptions with only invalid stacktraces', function() {
+    randomVal = 0.00000000000000001;
+    query.a = 0;
+    query.ca = 1;
+    query['3p'] = 0;
+    query.s = 'exception@file.js';
+    query.debug = 1;
+    query.r = 'referer';
+    query.m = 'message';
+    return chai.request(app).get('/r').query(query).then(function(res) {
+      throw new Error('Unreachable');
+    }, function(res) {
+      /** chai-http errors with handling > 299 status codes hence
+       * errors can only be asserted in the catch block which
+       * modifies anatomy of response
+       * object. More information at https://github.com/chaijs/chai-http/issues/75.
+       * This is a hack and once the package
+       * has been updated is subject to change
+       */
+      expect(res).to.have.property('status', statusCodes.BAD_REQUEST);
+      expect(res.response.text).to.equal('IGNORE');
+    });
+  });
 });
 
+describe('Test stacktrace conversions are done correctly', function() {
+  let testInput = [
+    `Error: localStorage not supported.
+    at Error (native)
+    at new vi (https://cdn.ampproject.org/rtv/031496877433269/v0.js:297:149)
+    at new  (https://cdn.ampproject.org/rtv/031496877433269/v0.js:298:365)
+    at dc (https://cdn.ampproject.org/rtv/031496877433269/v0.js:53:59)
+    at I (https://cdn.ampproject.org/rtv/031496877433269/v0.js:51:626)
+    at xi (https://cdn.ampproject.org/rtv/031496877433269/v0.js:298:278)
+    at mf.zc (https://cdn.ampproject.org/rtv/031496877433269/v0.js:408:166)
+    at pf (https://cdn.ampproject.org/rtv/031496877433269/v0.js:112:409)
+    at lf.$d (https://cdn.ampproject.org/rtv/031496877433269/v0.js:115:86)
+    at https://cdn.ampproject.org/rtv/031496877433269/v0.js:114:188`,
+    `Zd@https://cdn.ampproject.org/v0.js:5:204
+    error@https://cdn.ampproject.org/v0.js:5:314
+    jh@https://cdn.ampproject.org/v0.js:237:205
+    dc@https://cdn.ampproject.org/v0.js:53:69
+    G@https://cdn.ampproject.org/v0.js:51:510
+    ph@https://cdn.ampproject.org/v0.js:245:131
+    dc@https://cdn.ampproject.org/v0.js:53:69
+    gc@https://cdn.ampproject.org/v0.js:52:43
+    bh@https://cdn.ampproject.org/v0.js:226:461
+    dc@https://cdn.ampproject.org/v0.js:53:69
+    I@https://cdn.ampproject.org/v0.js:51:628
+    https://cdn.ampproject.org/v0.js:408:173
+    pf@https://cdn.ampproject.org/v0.js:112:411
+    $d@https://cdn.ampproject.org/v0.js:115:88
+    [native code]
+    https://cdn.ampproject.org/v0.js:115:170
+    promiseReactionJob@[native code]`,
+    `[native code]
+    https://cdn.ampproject.org/v0.js:115:170
+    promiseReactionJob@[native code]`,
+  ];
```

Comment:
Please use separate variables with descriptive names for each test input/output instead of arrays.